### DISCUSSION TITLE
Explore 'minified keys' toggle?

### DISCUSSION THREAD

**Discussion by zlovatt** [KraftyFX/aex#62] [Ideas]
This is a big conceptual idea and seems as if it'd require a lot of work to implement.

Currently, aex uses very verbose key names when serializing, opting for clarity and accuracy. See these footage interpretation keys:
- alphaMode
- conformFrameRate
- fieldSeparationType
- highQualityFieldSeparation
- invertAlpha
- loop
- premulColor
- removePulldown

For small datasets, this has a negligible impact on the output data size.

However, when serializing large AE projects / comps, this adds up; with one demo project, renaming all / many object key names to something shorter dropped the project considerably.
- `alphaMode` -> `aM`
- `conformFrameRate` -> `cFR`
- `fieldSeparationType` -> `fST`, etc

The downside is that we wouldn't _always_ want this mode on; often having long key names would make it very easy to target specific aex data for search & replace from external systems-- however, if the goal is file-size awareness (say, if somebody created a renderer for aex files, or they were distributed as part of an AE tool), this reduction in file size would help.

**Comment by rafikhan** [comment]
**Summary of our conversation from slack**
Minifying keys is a great way to reduce the size of the raw json.  This is very doable by building a pre-authored dictionary of name mappings so that it can go both ways. You'll probably get a decent bit of savings from this in the JSON string but the actual compressed (i.e. zipped) size probably won't change a huge amount.

This is roughly 1 week of dedicated heads down work and will require incremental maintenance any time AE introduces a new attribute that has to be serialized. 

This is valuable but not necessarily urgent.
